src/browser/browser.ts

```diff
@@ -230,9 +349,25 @@ export class Browser {
       this.log.debug('Taking screenshot', 'filePath', options.filePath);
     }
 
-    await this.withTimingMetrics<void>(() => {
-      return page.screenshot({ path: options.filePath });
-    }, 'screenshot');
+    const screenshotFn = () =>
+      page.screenshot({ path: options.filePath, fullPage: options.fullPageImage, captureBeyondViewport: options.fullPageImage });
+    await this.withTimingMetrics(
+      scrollResult.scrolled ? this.withFullPageViewport(screenshotFn, page, options, scrollResult.scrollHeight) : screenshotFn,
+      'screenshot'
+    );
+
+    if (options.scaleImage) {
+      const scaled = uniqueFilename(os.tmpdir()) + '.webp';
```

Comment:
@ArturWierzbicki -- I think we should change this back to .png for now.  Perhaps we can introduce a new 'format' parameter to change it to webp